File: examples/2015-day-01/swift/Tests/AoCTests/AoCTests.swift
import Testing

@testable import AoC

@Test(
  "Part One Examples",
  arguments: [
    (")", -1),
    ("()())", -1),
    ("(())", 0),
    ("()()", 0),
    ("(((", 3),
    ("(()(()(", 3),
    ("))(((((", 3),
    ("())", -1),
    ("))(", -1),
    (")))", -3),
    (")())())", -3),
  ]) func testPartOne(input: String, expected: Int)
{
  let actual = solvePartOne(input)

  #expect(expected == actual)
}

@Test(
  "Part Two Examples",
  arguments: [
    (")", 1),
    ("()())", 5),
  ]) func testPartTwo(input: String, expected: Int)
{
  let actual = solvePartTwo(input)

  #expect(expected == actual)
}

File: template/fsharp/tests/AocTests.fs
namespace Aoc.Tests

open Expecto

open Aoc

module AocTests =
    /// Checks one solution function against the example in the puzzle statement.
    let private checkSolution (solve : string -> int) (input : string) (expected : int) =
        let actual = solve input

        Expect.equal actual expected (sprintf "failed on input %A" input)

    [<Tests>]
    let aocTests =
        testList "Aoc" [
            testList "Part One" [ testCase "example" <| fun () -> checkSolution solvePartOne "" 0 ]

            testList "Part Two" [ testCase "example" <| fun () -> checkSolution solvePartTwo "" 0 ]
        ]
